src/ra.c

```diff
@@ -347,6 +351,104 @@ int ra_get_retransmit(void)
 	return ra_retransmit;
 }
 
+static bool ra_process_addr_eui64(void)
+{
+	struct ifreq ifr;
+	bool valid;
+	int sock;
+
+	sock = socket(AF_INET6, SOCK_DGRAM, 0);
+	if (sock < 0)
+		return false;
+
+	memset(&ifr, 0, sizeof(ifr));
+	strncpy(ifr.ifr_name, if_name, sizeof(ifr.ifr_name) - 1);
+
+	valid = !ioctl(sock, SIOCGIFHWADDR, &ifr);
+	close(sock);
+	if (!valid)
+		return false;
+
+	valid = odhcp6c_is_valid_ether_addr((uint8_t *) ifr.ifr_hwaddr.sa_data);
+	if (!valid)
+		return false;
+
+	lladdr.s6_addr[0] = 0xfe;
+	lladdr.s6_addr[1] = 0x80;
+	lladdr.s6_addr[8] = ifr.ifr_hwaddr.sa_data[0] ^ 0x2;
+	lladdr.s6_addr[9] = ifr.ifr_hwaddr.sa_data[1];
+	lladdr.s6_addr[10] = ifr.ifr_hwaddr.sa_data[2];
+	lladdr.s6_addr[11] = 0xff;
+	lladdr.s6_addr[12] = 0xfe;
+	lladdr.s6_addr[13] = ifr.ifr_hwaddr.sa_data[3];
+	lladdr.s6_addr[14] = ifr.ifr_hwaddr.sa_data[4];
+	lladdr.s6_addr[15] = ifr.ifr_hwaddr.sa_data[5];
+
+	return true;
+}
+
+static bool ra_process_addr_lla(void)
+{
+	struct sockaddr_in6 addr = {AF_INET6, 0, 0, ALL_IPV6_ROUTERS, if_index};
+	socklen_t alen = sizeof(addr);
+	bool valid;
+	int sock;
+
+	sock = socket(AF_INET6, SOCK_RAW, IPPROTO_ICMPV6);
+	if (sock < 0)
+		return false;
+
+	valid = !connect(sock, (struct sockaddr*) &addr, sizeof(addr)) &&
+		!getsockname(sock, (struct sockaddr*) &addr, &alen);
+	close(sock);
+	if (!valid)
+		return false;
+
+	lladdr = addr.sin6_addr;
+
+	return true;
+}
+
+static bool ra_process_addr_random(void)
+{
+	if (odhcp6c_random(&lladdr.s6_addr[8], 8) != 8)
+		return false;
+
+	lladdr.s6_addr[0] = 0xfe;
+	lladdr.s6_addr[1] = 0x80;
+
+	return true;
+}
+
+static void ra_process_addr(void)
+{
+	bool addr_lla = false;
+
+	switch (ra_ifid_mode) {
+	case RA_IFID_EUI64:
+		if (!ra_process_addr_eui64()) {
+			addr_lla = true;
+			syslog(LOG_ERR, "%s: error getting EUI64", if_name);
```

Comment:
Are we getting or setting or processing? We return true if we set this. Locating the error message is trivial for a dev, altho can we emit something more helpful to the user? Like "try xx instead"?

Reply:
Yeah. splitting these out would be good to help prevent conflated issues. (seems counter-intuitive to me that pppoEthernet  may not have a MAC :D )